mesonbuild/modules/python.py

```diff
@@ -297,6 +299,55 @@ def __init__(self, interpreter: 'Interpreter') -> None:
             'find_installation': self.find_installation,
         })
 
+    def _get_install_scripts(self) -> T.List[mesonlib.ExecutableSerialisation]:
+        backend = self.interpreter.backend
+        ret = []
+        installdata = backend.create_install_data()
+        py_files: T.Dict[str, T.List[T.Tuple[str, str]]] = collections.defaultdict(list)
+        optlevel = self.interpreter.environment.coredata.get_option(mesonlib.OptionKey('bytecompile', module='python'))
+        if optlevel == -1:
+            return ret
+
+        def should_append(f, isdir: bool = False):
+            return f.startswith(('{py_platlib}', '{py_purelib}')) and (f.endswith('.py') or isdir)
+
+        for t in installdata.targets:
+            if should_append(t.out_name):
+                py_files[t.subproject].append((t.out_name, os.path.join(installdata.prefix, t.outdir, os.path.basename(t.fname))))
```

Comment:
Why do you use the subproject as key in py_files? You don't seems to use that, can't be it be a flat list?

Reply:
I think this is leftover from trying to have the script only run for the subproject it is relevant to. I'm not sure that's needed anymore, I think I refactored it out after realizing that just always running the script and ignoring uninstalled files is fine, because the top-level project's scripts always run...